### ISSUE TITLE
Different fruit produce different amounts of wine when the number of kegs is the limiting factor.

### ISSUE THREAD

**Issue by Pluranium-Alloy** [Thorinair/Stardew-Profits#76]
Different fruit produce different amounts of wine when the number of kegs is the limiting factor.

![2024-06-11 (1)](https://github.com/Thorinair/Stardew-Profits/assets/136024471/e905e033-a34a-434b-8730-a389cb4a4405)
![2024-06-11](https://github.com/Thorinair/Stardew-Profits/assets/136024471/88ed838e-1e2f-40c6-8e21-fe1dfbea6e1d)

Both crops produce significant amounts of excess fruit yet, somehow, the calculator shows that the ancient fruit produces almost twice as much wine as the starfruit despite that. According to the wiki, the time taken to produce wine does not vary based on what fruit you use so, since starfruit is more valuable, it should be more profitable if the number of kegs is the limiting factor (limiting number of kegs can be useful to simulate a limited number of casks as one keg can supply 8 casks).

**Comment by Thorinair** [comment]
Indeed, this behavior is expected, because the equipment is applied as per-harvest, and ignores the time it takes to process the crops. I agree that this is a limitation.

Implementing code to account for processing times would get extremely complicated, as this time varies on time of day when the crop was placed into the equipment, and some other factors. See the note here: https://stardewvalleywiki.com/Keg#Products It would also be an additional layer of code complication as you would have to somehow simulate each crop hour by hour and how it is harvested/processed.

Because of this, there are currently no plans to implement time calculations, unless someone adventurous enough feels like writing this complex code. One alternative that could be done is to add an option to do one big processing at the end of the season, similar to the option that currently exists for Dehydrators, but I don't think anyone processes all of the fruit at the end of the season, so it was never really considered.

**Comment by IG2WTC** [comment]
Those are just some thoughts on how it might be possible to add it in an "easy" way (which will approximate some numbers but should be quite a bit more accurate than the current system). I might try to write something like that on the weekend.

1) Compromise and always assume the 3 day processing time. (When actually playing this will happen most of the time, so the actual output will only be slightly higher.)

2) Regarding our Jar processing capacity we could do something like this:
round down((Total days - Time to grow) / 3) = max possible Jar processes

3) Don´t account for Produce that will be done to late to be processed:
If no regrow:  total Produce - last Harvest = available Produce
Else: If regrow >= 3  ->  total Produce - last Harvest = available Produce
We can ignore faster regrows since those will have produced so much that we will have lots of excess anyway. (Unless we have extremely short durations, but then we will most certainly be limited by Jars.)

4) No we need to calculate much we can actually process:
If available Produce >= max possible Jars processes  ->  output max possible Jars processes
Else: output available Produce

With those rules we should be able to work for any Produce I can think of.
